assets/js/modules/analytics-4/components/settings/AdsConversionIDSettingsNotice.js

```diff
@@ -45,6 +47,35 @@ export default function AdsConversionIDSettingsNotice() {
 	const adsConversionIDMigratedAtMs = useSelect( ( select ) =>
 		select( MODULES_ANALYTICS_4 ).getAdsConversionIDMigratedAtMs()
 	);
+	const viewContext = useViewContext();
+	const trackDismissNotificationEvent = () => {
+		trackEvent(
+			`${ viewContext }_GA_Ads_redirect`,
+			'dismiss_notification'
+		);
+	};
+	const trackConfirmNotificationEvent = () => {
+		trackEvent(
+			`${ viewContext }_GA_Ads_redirect`,
+			'confirm_notification'
+		);
+	};
+
+	// Track a view_notification event.
+	useMount( () => {
+		// Only track the view event if the notice display condition is satisfied.
+		// The valid condition is a data migration within the 28 day window.
+		if (
+			adsConversionIDMigratedAtMs &&
+			Date.now() - adsConversionIDMigratedAtMs <
+				28 * DAY_IN_SECONDS * 1000
```

Comment:
As these conditions are now being repeated (to conditionally call `trackEvent` and render the notice), in the spirit of DRY, do you think it'd be wise to create a variable, e.g. `shouldShowNotice` and then use this in both places? LMK what you think, thanks!

Reply:
This was rolled into a single condition, thanks @nfmohit